src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -148,7 +154,38 @@ private BoundExpression BindDeconstruction(AssignmentExpressionSyntax node, Diag
                 hasErrors: hasErrors)
             { WasCompilerGenerated = true };
 
-            return new BoundDeconstructionAssignmentOperator(node, lhsTuple, boundConversion, returnType);
+            return new BoundDeconstructionAssignmentOperator(node, lhsTuple, boundConversion, resultIsUsed, returnType);
+        }
+
+        private static bool IsDeconstructionResultUsed(ExpressionSyntax left)
+        {
+            var parent = left.Parent;
+            if (parent is null || parent.Kind() == SyntaxKind.ForEachVariableStatement)
```

Comment:
Not yet. The `master` branch is using `2.3.0-beta2-61716-09` (from May 16th), but this optimization went into on May 30th. We will definitely refresh with RTM drop (if not sooner), and this PR is for 15.6.